sw/airborne/modules/checks/preflight_checks.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2022 Freek van Tienen <freek.v.tienen@gmail.com>
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/** 
+ * @file "modules/checks/preflight_checks.c"
+ * @author Freek van Tienen <freek.v.tienen@gmail.com>
+ * Adds preflight checks for takeoff
+ */
+
+#include "preflight_checks.h"
+#include "modules/datalink/telemetry.h"
+#include <stdio.h>
+
+/** Maximum combined message size for storing the errors */
+#ifndef PREFLIGHT_CHECK_MAX_MSGBUF
+#define PREFLIGHT_CHECK_MAX_MSGBUF 512
+#endif
+
+/** Seperating character for storing the errors */
+#ifndef PREFLIGHT_CHECK_SEPERATOR
+#define PREFLIGHT_CHECK_SEPERATOR ';'
+#endif
+
+/** Only send messages down every xx amount of seconds */
+#ifndef PREFLIGHT_CHECK_INFO_TIMEOUT
+#define PREFLIGHT_CHECK_INFO_TIMEOUT 5
+#endif
+
+static struct preflight_check_t *preflight_head = NULL;
+bool preflight_bypass = FALSE;
+
+/**
+ * @brief Register a preflight check and add it to the linked list
+ * 
+ * @param check The check to add containing a linked list
+ * @param func The function to register for the check
+ */
+void preflight_check_register(struct preflight_check_t *check, preflight_check_f func) {
+  // Prepend the preflight check
+  struct preflight_check_t *next = preflight_head;
+  preflight_head = check;
+  check->func = func;
+  check->next = next;
+}
+
+/**
+ * @brief Perform all the preflight checks
+ * 
+ * @return true When all preflight checks are successful
+ * @return false When one or more preflight checks fail
+ */
+bool preflight_check(void) {
```

Comment:
is this function called periodically somewhere on or only when set_motors_on is called ? I'm not sure